### PR TITLE
feat: support consul discovery

### PR DESCRIPTION
**Pull request by Fabriceli**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Support Consul discovery

**Related issues**

[feat #2695](https://github.com/apache/apisix-dashboard/issues/2695)

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by Fabriceli** [comment]
@guoqqqi @bzp2010 could you have a look at this PR, thanks

**Comment by Fabriceli** [comment]
return ci

**Comment by Fabriceli** [comment]
@bzp2010 Cloud you have a look ath this PR? Thanks

**Review by todorinskiz — APPROVED** [review]

**Comment by Fabriceli** [comment]
@bzp2010 @Baoyuantop Could you have a look at this PR?

**Comment by Baoyuantop** [comment]
> @bzp2010 @Baoyuantop Could you have a look at this PR?

Sorry for the late reply, I will review it later.

**Review by Baoyuantop — COMMENTED** [review]
Codes LGTM, we need to add an E2E test for it. You can refer to https://github.com/apache/apisix-dashboard/blob/master/web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js

**Comment by Fabriceli** [comment]
> Codes LGTM, we need to add an E2E test for it. You can refer to https://github.com/apache/apisix-dashboard/blob/master/web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js

Could you show me about how to run e2e test in local? Not found any command in Makefile

**Comment by Baoyuantop** [comment]
> > Codes LGTM, we need to add an E2E test for it. You can refer to https://github.com/apache/apisix-dashboard/blob/master/web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js
> 
> Could you show me about how to run e2e test in local? Not found any command in Makefile

We use Cypress for the E2E test, you can refer to https://docs.cypress.io/guides/getting-started/opening-the-app

**Comment by Fabriceli** [comment]
> > > Codes LGTM, we need to add an E2E test for it. You can refer to https://github.com/apache/apisix-dashboard/blob/master/web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js
> > 
> > 
> > Could you show me about how to run e2e test in local? Not found any command in Makefile
> 
> We use Cypress for the E2E test, you can refer to https://docs.cypress.io/guides/getting-started/opening-the-app

Please return CI to run the e2e test

**Comment by Fabriceli** [comment]
@Baoyuantop please talk a look at this PR, thanks

**Comment by Baoyuantop** [comment]
> @Baoyuantop please talk a look at this PR, thanks

Sorry for taking so long to reply, will deal with it today.

File: web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js [hunk 1]
```diff
@@ -154,6 +154,48 @@ context('Create Route with Service Discovery Upstream', () => {
...
+    cy.contains('Goto List').click();
+    cy.url().should('contains', 'routes/list');
+
+    // check if the changes have been saved
+    cy.get(selector.nameSelector).type(data.routeName);
+    cy.contains('Search').click();
+
+    cy.contains(data.routeName).siblings().contains('Configure').click();
+    // ensure it has already changed to edit page
+    cy.get(selector.name).should('value', data.routeName);
```
Comment (Baoyuantop) [inline web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js:192]:
```suggestion
    cy.get(selector.name).should('have.value', data.routeName);
```

File: web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js [hunk 2]
```diff
@@ -154,6 +154,48 @@ context('Create Route with Service Discovery Upstream', () => {
...
+    cy.contains('Next').click({
+      force: true,
+    });
+    cy.get(selector.service_name).should('value', `another.${data.serviceName}`);
```
Comment (Baoyuantop) [inline web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js:196]:
```suggestion
    cy.get(selector.service_name).should('have.value', `another.${data.serviceName}`);
```

Comment (Baoyuantop) [inline web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js:196]: Also need to check Discovery Type is Consul

File: web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js (hunk 1 above)
Comment (Fabriceli) [reply web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js]: done

File: web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js (hunk 2 above)
Comment (Fabriceli) [reply web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js]: done

Comment (Fabriceli) [reply web/cypress/e2e/route/create-route-with-service-discovery-upstream.cy.js]: done

**Comment by Fabriceli** [comment]
@Baoyuantop Please return CI, I add `consul` check, thanks

**Comment by Fabriceli** [comment]
Return CI

**Comment by Fabriceli** [comment]
@Baoyuantop Could you return the CI to re-run the workflows

**Comment by Fabriceli** [comment]
@Baoyuantop Fixed the CI error, please return CI, thanks

**Comment by Fabriceli** [comment]
@Baoyuantop take a look at this PR, thanks

**Review by Baoyuantop — APPROVED** [review]

**Comment by Fabriceli** [comment]
@guoqqqi @LiteSun @SkyeYoung take a look at this PR, thanks

**Comment by Fabriceli** [comment]
@Baoyuantop could you merge this PR, I have no permission to merge🙏

**Pull request merged**